[PL-58076]: SMP 0.22.0 RN (#8350)
harness/developer-hub: release-notes/self-managed-enterprise-edition.md

:::

:::info MongoDB Upgrade to Version 6.x in SMP Environments

> With MongoDB 5 reaching end-of-life (EOL), upgrading to MongoDB 6.x is recommended as an intermediate step for customers using an in-cluster MongoDB instance. This upgrade mitigates known CVEs and aligns with MongoDB’s recommended upgrade path.

Reviewer: ```suggestion
With MongoDB 5 reaching end-of-life (EOL), upgrading to MongoDB 6.x is recommended for customers using an in-cluster MongoDB instance. This upgrade mitigates known CVEs and aligns with MongoDB’s recommended upgrade path.
```

#### Upgrade Path

> 1. Upgrade to at least SMP 0.17.0 or if you are running 0.16.0 or below version. This is required because Mongo doesn’t allow direct upgrade from Mongo 4.0 to Mongo 6.0

Reviewer: ```suggestion
1. Upgrade to at least SMP 0.17.0 if you are running 0.16.0 or below version. This is required because Mongo doesn’t allow direct upgrade from Mongo 4.0 to Mongo 6.0
```

2. Upgrade to 0.22.0 or above.

#### Harness-Specific Details

1. **Helm Users:**
  - A job has been introduced to handle the FCV upgrade automatically. This job runs pre-upgrade checks and upgrades FCV to match the server version.

>   - The job is idempotent, meaning it will safely handle re-runs without unintended effects.

Reviewer: ```suggestion
```

>   - For installations using helm, customers should avoid direct MongoDB upgrades from version 4.4 to 6.0 to ensure compatibility.

Reviewer: ```suggestion
```

2. **ArgoCD Users:**
  - For ArgoCD users, run the FCV upgrade job manually before performing the main upgrade.

#### MongoDB FCV Upgrade Job

> To enable automatic FCV upgrade, include this job definition: (mongo-preupgradejob)[https://raw.githubusercontent.com/harness/helm-charts/refs/heads/main/src/harness/templates/mongo-preupgradejob.yaml]

Reviewer: ```suggestion
To run FCV upgrade manually (required for ArgoCD users), include this job definition: (mongo-preupgradejob)[https://raw.githubusercontent.com/harness/helm-charts/refs/heads/main/src/harness/templates/mongo-preupgradejob.yaml]
```

This job runs as a pre-upgrade hook to ensure FCV is compatible with the MongoDB server version.

#### Configuration

To disable the upgrade job if a manual FCV upgrade is preferred, use the following setting in the override file:

```yaml
upgrades:
  mongoFCVUpgrade:

>     enabled: true  # Set to false to disable

Reviewer: ```suggestion
    enabled: false
```

#### Continuous Integration

- Added support for setting Topology Spread Constraints to Kubernetes build pods. A new property, ‘podSpecOverlay’, has been introduced in the Kubernetes infrastructure properties within the CI stage, allowing users to apply additional settings to the build pod. Currently, this field supports specifying topologySpreadConstraint, with plans to extend support for additional configurations in the future. This feature requires using delegate version 24.09.83900 or higher (CI-14033)

- Added the ability to exclude connectors from the preflight check. This can be configured in the connector YAML by setting the property ignoreTestConnection to true. If the user sets this flag as true along with the feature flag CI_IGNORE_TEST_CONNECTION enabled, no matter the configuration, the connection test will always be marked as Successful. The feature is gated behind the feature flag CI_IGNORE_TEST_CONNECTION. (CI-13806, ZD-65275,65643)

#### Harness Platform

> 

Reviewer: Add the Mongo upgrade to 6.0 ticket RN

Author: done
